### PR TITLE
[MySQL Experience] (No.2) add link related entity classes.

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLLinkConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@ToString(exclude = {"password"})
+public class MySQLLinkConfig extends BaseLinkConfig {
+
+    // related to Azure MySQL
+    private Subscription subscription;
+    private Server server;
+    private DatabaseInner database;
+    private String username;
+    private char[] password;
+    // for persistence
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLLinkConfig.java:32]: merge as PasswordConfig

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/LinkPO.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private String moduleId;
+    private LinkType type;
+    private String envPrefix;
+
+    public LinkPO(String serviceId, String moduleId, LinkType type, String envPrefix) {
+        this.serviceId = serviceId;
+        this.moduleId = moduleId;
+        this.type = type;
+        this.envPrefix = envPrefix;
+    }
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/LinkPO.java:26]: @AllArg...

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLLinkConfig.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private PasswordConfig passwordConfig;
+    // ext
+    private String url;
+
+    public static MySQLLinkConfig getDefaultConfig(MySQLNode node) {
+        MySQLLinkConfig config = new MySQLLinkConfig();
+        if (Objects.nonNull(node)) {
+            Subscription subscription = AzureMvpModel.getInstance().getSubscriptionById(node.getSubscriptionId());
+            config.setSubscription(subscription);
+            config.setServer(node.getServer());
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLLinkConfig.java:40]: Could getServer() returns null?

**Review by Flanker32 — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLLinkConfig.java (hunk 3 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLLinkConfig.java:40]: it should not.

**Pull request merged**